Source/Diagnostics/WarpXOpenPMD.cpp

```diff
@@ -255,10 +254,70 @@ WarpXOpenPMDPlot::WriteOpenPMDParticles (MultiParticleContainer& mpc)
   }
 }
 
+void
+WarpXOpenPMDPlot::WriteOpenPMDParticles (const amrex::Vector<ParticleDiag>& particle_diags)
+{
+  WARPX_PROFILE("WarpXOpenPMDPlot::WriteOpenPMDParticles()");
+
+  for (unsigned i = 0, n = particle_diags.size(); i < n; ++i) {
+    WarpXParticleContainer* pc = particle_diags[i].getParticleContainer();
+    // names of amrex::Real and int particle attributes in SoA data
+    amrex::Vector<std::string> real_names;
+    amrex::Vector<std::string> int_names;
+    amrex::Vector<int> int_flags;
 
+    // see openPMD ED-PIC extension for namings
+    // note: an underscore separates the record name from its component
+    //       for non-scalar records
+    real_names.push_back("weighting");
+
+    real_names.push_back("momentum_x");
+    real_names.push_back("momentum_y");
+    real_names.push_back("momentum_z");
+
+    real_names.push_back("E_x");
+    real_names.push_back("E_y");
+    real_names.push_back("E_z");
+
+    real_names.push_back("B_x");
+    real_names.push_back("B_y");
+    real_names.push_back("B_z");
+
+#ifdef WARPX_DIM_RZ
+    real_names.push_back("theta");
+#endif
+    if(pc->DoFieldIonization()){
+       int_names.push_back("ionization_level");
+       // int_flags specifies, for each integer attribs, whether it is
+       // dumped as particle record in a plotfile. So far, ionization_level is the only
+       // integer attribs, and it is automatically dumped as particle record
+       // when ionization is on.
+       int_flags.resize(1, 1);
+    }
+
+    // Convert momentum to SI
+    pc->ConvertUnits(ConvertDirection::WarpX_to_SI);
+    // real_names contains a list of all real particle attributes.
+    // pc->plot_flags is 1 or 0, whether quantity is dumped or not.
```

Comment:
in the comment : pc->plot_flags will be particle_diags[i].plot_flags, since that is what is called below.